### ISSUE TITLE
Installer filenames contain "MacOSX" instead of correct "macOS" 

### ISSUE THREAD

**Issue by jezdez** [conda/constructor#849]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

Constructor seems to use an outdated name for the macOS platform that trickles down to the installers it builds.

We should start publishing our installers for “macOS” and not for “Mac OSX”, the latter hasn’t been accurate since 2016

**Comment by marcoesters** [comment]
What I am worried about here is that we currently do not know how many workflows will change. Other examples where these download links are used, homebrew, e.g.

**Comment by marcoesters** [comment]
For setup-miniconda, there will be the additional complication that you can specify a version number, so older installers will have the MacOSX moniker. So, whatever changes will be made there will have to deal with that name change.

**Comment by jaimergp** [comment]
Yes, we'll need to try both [platform names](https://github.com/conda-incubator/setup-miniconda/blob/e5293c8fd23bf30f6ad37a6281c8a924c8938440/src/constants.ts#L39-L43) til we find a valid download.
